src/Aura/Signal/Handler.php

```diff
@@ -115,6 +115,9 @@ public function exec($origin, $signal, array $args)
         // do the sender and signal match?
         if ($match_sender && $match_signal) {
             // yes, return an array of params with the callback return value
+            if (is_object($this->callback[0]) && get_class($this->callback[0]) == 'Aura\Di\LazyNew') {
```

Comment:
Currently we don't include other dependency inside a package. I probably want to look more into give you a solution though. I have not been using aura.signal in recent project.